craft_parts/plugins/rust_plugin.py

```diff
@@ -111,32 +154,109 @@ def get_build_snaps(self) -> Set[str]:
     @override
     def get_build_packages(self) -> Set[str]:
         """Return a set of required packages to install in the build environment."""
-        return {"gcc", "git"}
+        return {"curl", "gcc", "git", "pkg-config", "findutils"}
+
+    def _check_system_rust(self) -> bool:
+        """Check if Rust is installed on the system."""
+        try:
+            rust_version = subprocess.check_output(["rustc", "--version"])
+            cargo_version = subprocess.check_output(["cargo", "--version"])
+            return "rustc" in rust_version.decode(
+                "utf-8"
+            ) and "cargo" in cargo_version.decode("utf-8")
```

Comment:
This is a real nitpick, but I think it's more readable to do

```suggestion
            rust_version = subprocess.check_output(["rustc", "--version"], text=True)
            cargo_version = subprocess.check_output(["cargo", "--version"], text=True)
            return "rustc" in rust_version and "cargo" in cargo_version
```

Reply:
Done